packages/runtime-common/commands.ts

```diff
@@ -66,20 +68,38 @@ export abstract class Command<
     protected readonly configuration?: CommandConfiguration | undefined, // we'd like this to be required *if* CommandConfiguration is defined, and allow the user to skip it when CommandConfiguration is undefined
   ) {}
 
-  async execute(input: CardInputType): Promise<CardResultType> {
+  async execute(
+    input: CardInputType extends CardDef | undefined
+      ? CardInputType | Omit<CardInputType, keyof CardDef>
+      : never,
+  ): Promise<CardResultType> {
     // internal bookkeeping
     // todo: support for this.runTask being defined
     // runTask would be an ember task, run would just be a normal function
 
+    let inputCard: CardInputType;
+    if (input === undefined) {
+      inputCard = undefined as CardInputType;
+    } else if ('isCardDef' in input && input.isCardDef) {
+      inputCard = input as CardInputType;
+    } else {
+      let InputType = await this.getInputType();
+      if (!InputType) {
+        inputCard = undefined as CardInputType;
```

Comment:
perhaps this conditional branch should be an error?

Reply:
I think this side is for when there's no input required.

Reply:
no input is required but the user provided input -- sounds like a state we should be noisy about

Reply:
Oh of course, thanks, yeah this should throw an error